capstone/src/main/java/com/google/sps/data/RequestParser.java

```diff
@@ -22,4 +22,18 @@
         T object = gson.fromJson(sb.toString(), type);
         return object;
     }
+
+    public static String parseStringFromRequest(HttpServletRequest request) throws IOException {
+        BufferedReader reader = request.getReader();
+        StringBuilder sb = new StringBuilder();
+        try {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                sb.append(line).append('\n');
+            }
+        } finally {
+            reader.close();
+        }
+        return sb.toString();
+    }
```

Comment:
Keeping these separate creates code duplication. Instead, the more complicated version of the method (parseObject) should delegate to the simpler version:
public static <T> T parseObjectFromRequest(HttpServletRequest request, Class<T> type) throws IOException {
        Gson gson = new Gson();
        T object = gson.fromJson(parseStringFromRequest(request), type);
        return object;
    }

Also see my comment on the original PR #5 about this method